### PR TITLE
Issue 51557: Avoid folder deletion summary stats type casting

### PR DESCRIPTION
**Pull request by labkey-nicka**
#### Rationale
This is a followup to #6065 where the fix to casting switched from `Long` to `Integer`. That worked great on SQL Server, however, caused the same problem on Postgres instances. This fix avoids calling `TableSelector.getValueMap()` all together as it provides inconsistent type instances depending on the database type.

#### Related Pull Requests
- #6005 
- #6065

#### Changes
- Refactor folder summary stats for data class and sample type data counts to use `getResultSet()` instead of `getValueMap()`.

### CODE REVIEW

**Comment by labkey-nicka** [comment]
Test suites:

- [BiologicsFolderAdminTest](https://teamcity.labkey.org/buildConfiguration/LabKey_2411_Premium_ProductSuites_Biologics_BiologicsAPostgres/3272710)
- [SMProFolderAdminFromChildFolderTest](https://teamcity.labkey.org/buildConfiguration/LabKey_2411_Premium_ProductSuites_SampleManager_SampleManagerBPostgres/3272724) (PG)
- [SMProFolderAdminFromChildFolderTest](https://teamcity.labkey.org/buildConfiguration/LabKey_2411_Premium_ProductSuites_SampleManager_SampleManagerBSqlserver/3272712) (MSSQL)

File: experiment/src/org/labkey/experiment/ExperimentModule.java [hunk 1]
```diff
@@ -889,12 +892,21 @@ public Collection<String> getSummary(Container c)
             columns.add(ExpDataClassTable.Column.Name.name());
             columns.add(ExpDataClassTable.Column.DataCount.name());
 
-            Map<String, Integer> results = new TableSelector(table, columns).getValueMap();
```
Comment (labkey-jeckels) [inline experiment/src/org/labkey/experiment/ExperimentModule.java:892]: This refactor seems fine. As an alternative approach, I've used `Map<String, Number>` in similar scenarios.

**Review by labkey-jeckels — APPROVED** [review]

File: experiment/src/org/labkey/experiment/ExperimentModule.java (hunk 1 above)
Comment (labkey-nicka) [reply experiment/src/org/labkey/experiment/ExperimentModule.java:892]: That seems reasonable, however, doesn't that just push the problem to the next statement?

Comment (labkey-jeckels) [reply experiment/src/org/labkey/experiment/ExperimentModule.java:892]: You can call `Number.intValue()` or `Number.longValue()`

**Review by labkey-susanh — APPROVED** [review]

File: experiment/src/org/labkey/experiment/ExperimentModule.java (hunk 1 above)
Comment (labkey-nicka) [reply experiment/src/org/labkey/experiment/ExperimentModule.java:892]: I've tested locally and revised to this approach. Thanks for the tip.

**Pull request merged**